PowerEditor/src/NppCommands.cpp

```diff
@@ -3122,44 +3141,95 @@ void Notepad_plus::command(int id)
 
 			if (idEncoding != -1)
 			{
-				// Save the current clipboard content
-				::OpenClipboard(_pPublicInterface->getHSelf());
-				HANDLE clipboardData = ::GetClipboardData(CF_TEXT);
-				LPVOID clipboardData2 = NULL;
-				if (clipboardData != NULL)
+				// try to save the current clipboard CF_TEXT content 1st
+				HGLOBAL hglbClipboardCopy = NULL;
+				if (::OpenClipboard(_pPublicInterface->getHSelf()))
 				{
-					int len = static_cast<int32_t>(::GlobalSize(clipboardData));
-					LPVOID clipboardDataPtr = ::GlobalLock(clipboardData);
-
-					HANDLE allocClipboardData = ::GlobalAlloc(GMEM_MOVEABLE, len);
-					clipboardData2 = ::GlobalLock(allocClipboardData);
-
-					::memcpy(clipboardData2, clipboardDataPtr, len);
-					::GlobalUnlock(clipboardData);
-					::GlobalUnlock(allocClipboardData);
+					HANDLE hClipboardData = ::GetClipboardData(CF_TEXT);
+					if (hClipboardData) // NULL if there is no previous CF_TEXT data in
+					{
+						LPVOID pClipboardData = ::GlobalLock(hClipboardData);
+						if (pClipboardData)
+						{
+							size_t clipboardDataSize = ::GlobalSize(pClipboardData);
+							hglbClipboardCopy = ::GlobalAlloc(GMEM_MOVEABLE, clipboardDataSize);
+							if (hglbClipboardCopy)
+							{
+								LPVOID pClipboardCopy = ::GlobalLock(hglbClipboardCopy);
+								if (pClipboardCopy)
+								{
+									::memcpy(pClipboardCopy, pClipboardData, clipboardDataSize);
+									::GlobalUnlock(hglbClipboardCopy);
+								}
+								else
+								{
+									::GlobalFree(hglbClipboardCopy);
+									hglbClipboardCopy = NULL;
+								}
+							}
+							::GlobalUnlock(hClipboardData);
+						}
+					}

```

Comment:
@donho 
>There are many levels of depth here. That makes hard to read the code.
OTOH, if hglbClipboardCopy is NULL, the operation should be halted.

I understand the request (I tried to follow exactly this in other places of the patch) but in this case it is valid to continue in execution of this part of code, even if one cannot store the previous Clipboard content to the hglbClipboardCopy. This will e.g. happen if the previous Clipboard data is so large (or the free memory is so much fragmented) that the GlobalAlloc fails (but it does not automatically mean that the following encoding conversion steps using the Clipboard will fail too).

I added that more indented code part there also because it is relatively short and clear...

Reply:
@xomx 
By reading again my code (which has been done since long time ago), I do remember the logic now.
You're right about it!